sys/net/gnrc/application_layer/dhcpv6/client_6lbr.c

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) 2018-20 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @author  Martine S. Lenders <m.lenders@fu-berlin.de>
+ */
+
+#include <assert.h>
+
+#include "event.h"
+#include "log.h"
+#include "net/dhcpv6/client.h"
+#include "net/ipv6/addr.h"
+#include "net/gnrc.h"
+#include "net/gnrc/ipv6/nib/ft.h"
+#include "net/gnrc/netif/internal.h"
+
+#include "net/gnrc/dhcpv6/client/6lbr.h"
+
+static char _stack[DHCPV6_CLIENT_STACK_SIZE];
+
+static gnrc_netif_t *_find_upstream_netif(void)
+{
+    gnrc_netif_t *netif = NULL;
+
+    if (CONFIG_GNRC_DHCPV6_CLIENT_6LBR_UPSTREAM) {
+        return gnrc_netif_get_by_pid(CONFIG_GNRC_DHCPV6_CLIENT_6LBR_UPSTREAM);
+    }
+    while ((netif = gnrc_netif_iter(netif))) {
+        if (!gnrc_netif_is_6lo(netif)) {
+            LOG_WARNING("DHCPv6: Selecting interface %d as upstream\n",
+                        netif->pid);
+            return netif;
+        }
+    }
+    return NULL;
+}
+
+static void _configure_upstream_netif(gnrc_netif_t *upstream_netif)
+{
+    ipv6_addr_t addr = {
+            .u8 = { 0xfe, 0x80, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1 }
+        };
+
+    /* set default route to host machine (as set-up in setup_network.sh) */
+    gnrc_ipv6_nib_ft_add(NULL, 0, &addr, upstream_netif->pid, 0);
+    /* set additional link-local address to allow for easy static route
+     * configuration */
+    addr.u8[15] = 2;
+    gnrc_netif_ipv6_addr_add(upstream_netif, &addr, 64, 0);
+                /* configure client to request prefix delegation for WPAN
+                 * interface */
+}
```

Comment:
Considering the delegating router is configured properly (sending RAs to advertise themselves as router, set downstream routes to clients the prefix was delegated to), this might not be necessary, but I kept it in, just to be safe.

Reply:
This is only really useful for testing (when your machine is the next hop), for real deployments this should not be needed.
Even worse, it looks like this just fills up the 'slots' for possible IP addresses for an interface that we allocate by default.

```
2020-02-16 22:46:44,231 # Iface  9  HWaddr: 3C:71:BF:9E:13:FC  Link: up 
2020-02-16 22:46:44,235 #           L2-PDU:1500 MTU:1492  HL:255  RTR  
2020-02-16 22:46:44,239 #           RTR_ADV  Source address length: 6
2020-02-16 22:46:44,241 #           Link type: wireless
2020-02-16 22:46:44,247 #           inet6 addr: fe80::3e71:bfff:fe9e:13fc  scope: link  VAL
2020-02-16 22:46:44,254 #           inet6 addr: 2001:16b8:4554:2b00:3e71:bfff:fe9e:13fc  scope: global  VAL
2020-02-16 22:46:44,257 #           inet6 group: ff02::2
2020-02-16 22:46:44,259 #           inet6 group: ff02::1
2020-02-16 22:46:44,263 #           inet6 group: ff02::1:ff9e:13fc
2020-02-16 22:46:44,264 #           
2020-02-16 22:46:44,269 # Iface  7  HWaddr: 39:B2  Channel: 26  Page: 0  NID: 0x23
2020-02-16 22:46:44,273 #           Long HWaddr: AA:1F:52:9C:BD:30:DF:AA 
2020-02-16 22:46:44,280 #            TX-Power: 0dBm  State: IDLE  max. Retrans.: 3  CSMA Retries: 4 
2020-02-16 22:46:44,285 #           AUTOACK  ACK_REQ  CSMA  L2-PDU:102 MTU:1280  HL:64  RTR  
2020-02-16 22:46:44,288 #           RTR_ADV  6LO  IPHC  
2020-02-16 22:46:44,291 #           Source address length: 8
2020-02-16 22:46:44,294 #           Link type: wireless
2020-02-16 22:46:44,300 #           inet6 addr: fe80::a81f:529c:bd30:dfaa  scope: link  VAL
2020-02-16 22:46:44,307 #           inet6 addr: 2001:16b8:4554:2bf8:a81f:529c:bd30:dfaa  scope: global  VAL
2020-02-16 22:46:44,309 #           inet6 group: ff02::2
2020-02-16 22:46:44,312 #           inet6 group: ff02::1
2020-02-16 22:46:44,316 #           inet6 group: ff02::1:ff30:dfaa
2020-02-16 22:46:44,317 #           
2020-02-16 22:47:08,978 #  ifconfig 9 add unicast fe80::2
2020-02-16 22:47:08,983 # error: unable to add IPv6 address
```

So with this we will never get a proper address even if the upstream router is correctly configured.